cmd/start.go

```diff
@@ -267,30 +273,30 @@ func initializeBalanceMonitoring(
 	ctx context.Context,
 	chainProvider chain.Handle,
 	config *config.Config,
-	ethereumAddress string,
+	ethereumKey *keystore.Key,
 ) {
-	balanceMonitor, err := chainProvider.BalanceMonitor()
+	balanceMonitor, err := chainProvider.(*ethereum.EthereumChain).BalanceMonitor()
```

Comment:
This change is contradicting with @Shadowfiend's approach presented in https://github.com/keep-network/keep-ecdsa/pull/693/. In the referenced PR we made `EthereumChain` not exported and `ethereum.Connect` function returns `chain.Handle` instead of `EthereumChain`.

If we consider `BalanceMonitor` Ethereum-specific concept, and it indeed makes sense and was noted in the referenced PR, we need to find better way to instantiate it than having to cast `chain.Handle`.

One option is to have `ethereum.Connect` return `chain.Handle` interface and `ethereum.BalanceMonitor`. What do you think about it?

Reply:
This is just a temporary workaround. Once we'll do the multi-chain refactor this one will disappear from `start.go` entirely. However, I agree it can look better. Please see: https://github.com/keep-network/keep-core/commit/89f10b78f353c33ad427d4f7215725039126805b

Reply:
We will still need to have a way to create the balance monitor and even though all of this code will be in Ethereum-specific connector, we will still have the same problem, nope?

Reply:
We can make the `ethereum.Connect` return `*ethereumChain` and use it only in a specific connector function which will return `chain.Handle`. We do this in `keep-ecdsa` https://github.com/keep-network/keep-ecdsa/blob/celo/cmd/connector_ethereum.go. Then we won't need type assertion in `ethereum.BalanceMonitor`.

Reply:
Unless we move the connector to `ethereum` package, we run into a conflict with the approach from https://github.com/keep-network/keep-ecdsa/pull/693 because we'll have to make `ethereumChain` exported and return it from the `Connect` function instead of `chain.Handle`.

Reply:
Please see https://github.com/keep-network/keep-core/commit/5ac723255980b02ad039e199108b0de8202e2ece